File: src/main/java/amerifrance/guideapi/objects/Category.scala
package amerifrance.guideapi.objects

import java.util

import amerifrance.guideapi.gui.GuiBase
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.item.ItemStack
import net.minecraft.util.StatCollector

class Category(entryList: util.List[Entry] = new util.ArrayList[Entry](), unlocCategoryName: String, itemstack: ItemStack) {

  var entries: util.List[Entry] = entryList
  var unlocalizedCategoryName: String = unlocCategoryName
  var stack = itemstack

  def addEntry(entry: Entry) = {
    this.entries.add(entry)
  }

  def addEntryList(list: util.List[Entry]) = {
    this.entries.addAll(list)
  }

  def removeEntry(entry: Entry) = {
    this.entries.remove(entry)
  }

  def removeCategoryList(list: util.List[Entry]) = {
    this.entries.remove(list)
  }

  def getLocalizedName(): String = {
    return StatCollector.translateToLocal(unlocalizedCategoryName)
  }

  def drawExtras(mouseX: Int, mouseY: Int, guiBase: GuiBase) = {
  }

  def canSee(player: EntityPlayer): Boolean = {
    return true
  }

  def onLeftClicked(mouseX: Int, mouseY: Int) = {
    System.out.println(getLocalizedName() + "Left Clicked")
  }

  def onRightClicked(mouseX: Int, mouseY: Int) = {
    System.out.println(getLocalizedName() + "Right Clicked")
  }
}

File: src/main/java/de/maxanier/guideapi/api/impl/Category.java
package de.maxanier.guideapi.api.impl;

import de.maxanier.guideapi.api.impl.abstraction.CategoryAbstract;
import de.maxanier.guideapi.api.impl.abstraction.EntryAbstract;
import de.maxanier.guideapi.gui.BaseScreen;
import de.maxanier.guideapi.gui.CategoryScreen;
import de.maxanier.guideapi.gui.HomeScreen;
import net.minecraft.client.Minecraft;
import net.minecraft.client.gui.GuiGraphics;
import net.minecraft.network.chat.Component;
import net.minecraft.resources.Identifier;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.item.ItemStack;
import net.neoforged.api.distmarker.Dist;
import net.neoforged.api.distmarker.OnlyIn;

import java.util.Map;

public class Category extends CategoryAbstract {

    public Category(Map<Identifier, EntryAbstract> entryList, Component name) {
        super(entryList, name);
    }

    public Category(Component name) {
        super(name);
    }

    @Override
    public boolean canSee(Player player, ItemStack bookStack) {
        return true;
    }

    @Override
    public void draw(GuiGraphics graphics, Book book, int categoryX, int categoryY, int categoryWidth, int categoryHeight, int mouseX, int mouseY, BaseScreen guiBase, boolean drawOnLeft) {
    }

    @Override
    public void drawExtras(GuiGraphics graphics, Book book, int categoryX, int categoryY, int categoryWidth, int categoryHeight, int mouseX, int mouseY, BaseScreen guiBase, boolean drawOnLeft) {
    }

    @Override
    public void onInit(Book book, HomeScreen guiHome, Player player, ItemStack bookStack) {
    }

    @Override
    public void onLeftClicked(Book book, double mouseX, double mouseY, Player player, ItemStack bookStack) {
        Minecraft.getInstance().setScreen(new CategoryScreen(book, this, player, bookStack, null));
    }

    @Override
    public void onRightClicked(Book book, double mouseX, double mouseY, Player player, ItemStack bookStack) {
    }
}
